core/Period/Month.php

```diff
@@ -83,15 +83,15 @@ protected function generate()
     protected function processOptimalSubperiods($startDate, $endDate)
     {
         while ($startDate->isEarlier($endDate)
-            || $startDate == $endDate
+            || $startDate === $endDate
```

Comment:
I don't think that will work, as both variables are an instance of `Piwik\Date` but aren't the same instance. This might work though
```suggestion
            || $startDate->toString() === $endDate->toString()
```